File: models/recipe.py
from dataclasses import dataclass, field
from typing import List, Optional


@dataclass
class Category:
    id: Optional[int]
    name: str
    is_builtin: bool = False


@dataclass
class RecipeIngredient:
    """An ingredient line attached to a recipe (quantity + unit)."""

    id: Optional[int]
    ingredient_id: Optional[int]
    name: str
    quantity: float = 0.0
    unit: str = ""


@dataclass
class Recipe:
    id: Optional[str]
    title: str
    steps: str
    cook_time_minutes: int = 0
    category_id: Optional[int] = None
    category_name: str = ""
    photo_path: Optional[str] = None
    created_at: Optional[str] = None
    updated_at: Optional[str] = None
    ingredients: List[RecipeIngredient] = field(default_factory=list)

File: android/app/src/main/java/com/kapkabi/helpmeneger/domain/model/Recipe.kt
package com.kapkabi.helpmeneger.domain.model

data class Recipe(
    val id: String,
    val title: String,
    val steps: String = "",
    val cookTimeMinutes: Int = 0,
    val categoryId: Long? = null,
    val categoryName: String = "",
    val photoPath: String? = null,
    val createdAt: String? = null,
    val updatedAt: String? = null,
    val ingredients: List<RecipeIngredient> = emptyList(),
)

data class RecipeSummary(
    val id: String,
    val title: String,
    val cookTimeMinutes: Int,
    val photoPath: String?,
    val createdAt: String,
    val updatedAt: String,
    val categoryId: Long?,
    val categoryName: String?,
)

enum class RecipeSortField { TITLE, COOK_TIME, DATE }

enum class SortOrder { ASC, DESC }
